File: Frontend/src/app/models/SellerDetail.ts
export interface SellerDetail {
    id:number,
    firstName:string;
    fatherName:string;
    grandFather:string;
    electronicNationalIdNumber?:string;
    phoneNumber:string;
    paddressProvinceId:Number;
    paddressDistrictId:number;
    paddressVillage:string;
    taddressProvinceId:number;
    taddressDistrictId:number;
    taddressVillage:string;
    propertyDetailsId:number;
    photo:string;
    nationalIdCard?:string;
    nationalIdCardPath?:string;
    roleType:string; // Seller role type
    authorizationLetter?:string; // Path to authorization letter file (for agents)
    heirsLetter?:string; // Path to heirs letter file (for heirs)
    propertyTypeId?:number;
    customPropertyType?:string;
    price?:number;
    priceText?:string;
    royaltyAmount?:number;
    halfPrice?:number;
    rentStartDate?:string; // Rental start date as Hijri Shamsi string
    rentEndDate?:string; // Rental end date as Hijri Shamsi string
    rentStartDateStr?:string; // Rent start date string for backend
    rentEndDateStr?:string; // Rent end date string for backend
    calendarType?:string; // Calendar type for date conversion
    transactionType?:string; // Transaction type (Purchase, Rent, Revocable Sale, Other)
    transactionTypeDescription?:string; // Custom description when transactionType is 'Other'
    taxIdentificationNumber?:string;
    additionalDetails?:string;
    sharePercentage?:number;
    shareAmount?:number;
}

export interface VBuyerDetail {
    id:number,
    firstName:string;
    fatherName:string;
    grandFather:string;
    electronicNationalIdNumber?:string;
    phoneNumber:string;
    paddressProvinceId:Number;
    paddressDistrictId:number;
    paddressVillage:string;
    taddressProvinceId:number;
    taddressDistrictId:number;
    taddressVillage:string;
    propertyDetailsId:number;
    photo:string;
    nationalIdCard?:string;
    nationalIdCardPath?:string;
    roleType:string; // Seller role type
    authorizationLetter?:string; // Path to authorization letter file (for agents)
    heirsLetter?:string; // Path to heirs letter file (for heirs)
    propertyTypeId?:number;
    customPropertyType?:string;
    price?:number;
    priceText?:string;
    royaltyAmount?:number;
    halfPrice?:number;
    rentStartDate?:string; // Rental start date as Hijri Shamsi string
    rentEndDate?:string; // Rental end date as Hijri Shamsi string
    rentStartDateStr?:string; // Rent start date string for backend
    rentEndDateStr?:string; // Rent end date string for backend
    calendarType?:string; // Calendar type for date conversion
    transactionType?:string; // Transaction type (Purchase, Rent, Revocable Sale, Other)
    transactionTypeDescription?:string; // Custom description when transactionType is 'Other'
    sharePercentage?:number;
    shareAmount?:number;
}

File: Backend/Models/Property/SellerDetail.cs
using System;
using System.Collections.Generic;

namespace WebAPIBackend.Models;

public partial class SellerDetail
{
    public int Id { get; set; }

    public string FirstName { get; set; } = null!;

    public string FatherName { get; set; } = null!;

    public string GrandFather { get; set; } = null!;

    // Electronic National ID - الیکټرونیکی تذکره
    public string? ElectronicNationalIdNumber { get; set; }

    public string? PhoneNumber { get; set; }

    public int? PaddressProvinceId { get; set; }

    public int? PaddressDistrictId { get; set; }

    public string? PaddressVillage { get; set; }

    public int? TaddressProvinceId { get; set; }

    public int? TaddressDistrictId { get; set; }

    public string? TaddressVillage { get; set; }

    public int? PropertyDetailsId { get; set; }

    public DateTime? CreatedAt { get; set; }

    public string? CreatedBy { get; set; }

    public string? Photo { get; set; }

    public string? NationalIdCard { get; set; }

    public string? RoleType { get; set; } // "Seller" or "Authorized Agent (Seller)"

    public string? AuthorizationLetter { get; set; } // Path to authorization letter file

    public string? HeirsLetter { get; set; } // Path to heirs letter file

    public string? TaxIdentificationNumber { get; set; }

    public string? AdditionalDetails { get; set; }

    public string? Price { get; set; }

    public string? RoyaltyAmount { get; set; }

    public string? HalfPrice { get; set; }

    public string? SharePercentage { get; set; }

    public string? ShareAmount { get; set; }

    public virtual Location? PaddressDistrict { get; set; }

    public virtual Location? PaddressProvince { get; set; }

    public virtual PropertyDetail? PropertyDetails { get; set; }

    public virtual Location? TaddressDistrict { get; set; }

    public virtual Location? TaddressProvince { get; set; }
    public virtual ICollection<Propertyselleraudit> Propertyselleraudits { get; } = new List<Propertyselleraudit>();
}